pkg/agent/openflow/pipeline.go

```diff
@@ -927,15 +993,22 @@ func (c *client) gatewayIPSpoofGuardFlows(gatewayOFPort uint32, hasIPv6Addr bool
 }
 
 // serviceCIDRDNATFlow generates flows to match dst IP in service CIDR and output to host gateway interface directly.
-func (c *client) serviceCIDRDNATFlow(serviceCIDR *net.IPNet, gatewayMAC net.HardwareAddr, gatewayOFPort uint32) binding.Flow {
-	return c.pipeline[dnatTable].BuildFlow(priorityNormal).MatchProtocol(binding.ProtocolIP).
-		MatchDstIPNet(*serviceCIDR).
-		Action().SetDstMAC(gatewayMAC).
-		Action().LoadRegRange(int(portCacheReg), gatewayOFPort, ofPortRegRange).
-		Action().LoadRegRange(int(marksReg), portFoundMark, ofPortMarkRange).
-		Action().GotoTable(conntrackCommitTable).
-		Cookie(c.cookieAllocator.Request(cookie.Service).Raw()).
-		Done()
+func (c *client) serviceCIDRDNATFlow(serviceCIDRs []*net.IPNet, gatewayMAC net.HardwareAddr, gatewayOFPort uint32) []binding.Flow {
```

Comment:
Better to rename the function to serviceCIDRDNATFlows.

Reply:
Done.